docs/ADR-49-signed-fetch-v2.md

```diff
@@ -0,0 +1,365 @@
+
+Extra metadata send to the server (encoded as [JSON](https://datatracker.ietf.org/doc/html/rfc4627))
+
+```ts
+  Metadata = `{"catalyst": "peer.decentraland.org"}`
+```
+
+#### 4.5. ExtraHeaders
+
+> Optional
+
+To sign other headers not listed previously Users can include them usign `X-Identity-Headers`, which is a list separated by semicolon. If this header all extra headers listed *MUST* be included in the sign as well in the same order they were listed
+
+```ts
+  ExtraHeaders = "x-identity-headers:" +
+    LOWERCASE(SIGNED_HEADER_1) + ';' + LOWERCASE(SIGNED_HEADER_2) + ';' + LOWERCASE(SIGNED_HEADER_N) + '\n' +
+    LOWERCASE(SIGNED_HEADER_1)  + ':' + TRIM(Headers[SIGNED_HEADER_1]) + '\n' +
+    LOWERCASE(SIGNED_HEADER_2)  + ':' + TRIM(Headers[SIGNED_HEADER_2]) + '\n' +
+    LOWERCASE(SIGNED_HEADER_N)  + ':' + TRIM(Headers[SIGNED_HEADER_N])
+```
+
+> Example: sign `Accept` and `Cookie` headers
+>
+> ```ts
+>   ExtraHeaders = 'x-identity-headers:accept;cookie\n' +
+>     'accept:application/json\n' +
+>     'cookie:lang=en'
+> ```
+
+### 5. BodyHashPayload
+
+> Required if: the request includes some body content it *MUST* be present, otherwise it *MUST* be omitted
+
+For almost every cases this is the result of hashing the content of the body, but for [`multipart/form-data` request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#content-type_in_html_forms) a different approach is needed in order to be used on web applications
+
+When `ContentType != 'multipart/form-data'`
+
+### 5.1 ContentType != multipart/form-data
+
+Just hash the entire body
+
+```ts
+  BodyHashPayload = SHA265(REQUEST_BODY)
+```
+
+### 5.2 ContentType == multipart/form-data
+
+The ordered list of all fields in the request, each field *MUST* be normalized as well
+
+```ts
+  BodyHashPayload = SORT(CanonicalField1, CanonicalField2 /*, ... */)
+```
+
+Each field *MUST* include `name` and `size` directives as prefix of the content hash
+
+```ts
+  CanonicalFieldX = 'name="description";' +
+    'size=50;' +
+    SHA256(FieldContent)
+```
+
+Additionally if some fields are files they *MUST* also include `filename` and `type` directives
+
+```ts
+  CanonicalFieldX = 'name="description";' +
+    'filename="image.png";' +
+    'type="image/png";' +
+    'size=99999999999;' +
+    SHA256(FieldContent)
+
+```
+
+### Canonical Request Examples
+
+#### `GET https://decentraland.org/api/status`
+
+```txt
+GET /api/status\n
+host:decentraland.org\n
+x-identity-expiration:2020-01-01T00:00:00Z
+```
+
+#### `GET https://decentraland.org/api/status` with metadata
+
+```txt
+GET /api/status
+host:decentraland.org\n
+x-identity-expiration:2020-01-01T00:00:00Z
+x-identity-metadata:{"service":"market.decentraland.org"}
+```
+
+#### `POST https://decentraland.org/api/status?filter=asc` with metadata
+
+```txt
+POST /api/status?filter=asc\n
+host:decentraland.org\n
+x-identity-expiration:2020-01-01T00:00:00Z
```

Comment:
tailin \n is not consistent here, should \n be explicit in this representation? we already have new lines between lines

Reply:
it is just a visual help

Reply:
we can delete if it seems confusing

Reply:
yes please remove it if it is only a clue